File: app/src/main/java/com/example/pam_app/view/AddIncomeView.java
package com.example.pam_app.view;

import com.example.pam_app.model.Income;

public interface AddIncomeView {

    void onErrorSavingIncome();
    void onSuccessSavingIncome(final Income income);
    void showDescriptionError(final int error, final Integer parameter);
    void showAmountError(final int error, final Integer parameter);
    void showDateError(final int error);
}

File: app/src/main/java/com/example/pam_app/view/AddIncomeView.kt
package com.example.pam_app.view

import com.example.pam_app.model.Income
import kotlin.jvm.Synchronized

interface AddIncomeView {
    fun onErrorSavingIncome()
    fun onSuccessSavingIncome(income: Income)
    fun showDescriptionError(error: Int, parameter: Int?)
    fun showAmountError(error: Int, parameter: Int?)
    fun showDateError(error: Int)
}